File: src/main/java/app/askresume/domain/prompt/repository/PromptRepository.java
package app.askresume.domain.prompt.repository;

import app.askresume.domain.prompt.constant.PromptType;
import app.askresume.domain.prompt.model.Prompt;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface PromptRepository extends JpaRepository<Prompt, Long> {

    Optional<Prompt> findByPromptType(PromptType promptType);
}

File: src/main/kotlin/app/askresume/domain/prompt/repository/PromptRepository.kt
package app.askresume.domain.prompt.repository

import app.askresume.domain.prompt.constant.PromptType
import app.askresume.domain.prompt.exception.PromptNotFoundException
import app.askresume.domain.prompt.model.Prompt
import org.springframework.data.jpa.repository.JpaRepository

fun PromptRepository.findPromptByPromptType(promptType: PromptType) : Prompt =
    findByPromptType(promptType) ?: throw PromptNotFoundException(promptType.name)

interface PromptRepository : JpaRepository<Prompt, Long> {
    fun findByPromptType(promptType: PromptType): Prompt?
}

